### PR TITLE
Dev-mode: don't warn if multi-module project

### LINKED ISSUE #9093: Dev-Mode: Remove "[WARNING] The project's sources directory does not exist"
**Description**
In the process of migrating my current project from 1.3.1 to 1.4.2, I happily removed  a `Dummy.java` from the main module after having seen this comment: https://github.com/quarkusio/quarkus/pull/8712#issuecomment-616989898

This now yields a warning:
```
[WARNING] The project's sources directory does not exist C:\Develop\SomeProject\server\src\main\java
```

I suggest to remove this warning, especially for multi-module projects.

cc @harthorst

### PR DESCRIPTION
**Pull request by famod**
Fixes #9093

@aloubyansky Funnily, the `projects/multimodule` IT project was already using the "java-src-less" setup I outlined in the issue. 😁 

Btw, `projects/classic-no-java-src` is a copy of `projects/classic`, but without `src/main/java`.

### CODE REVIEW

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 1]
```diff
@@ -265,7 +266,8 @@ public void execute() throws MojoFailureException, MojoExecutionException {
             return;
         }
 
-        if (!sourceDir.isDirectory()) {
+        // don't warn if running in a module of a multi-module project
+        if (!sourceDir.isDirectory() && !Optional.ofNullable(project.getParentFile()).map(File::exists).orElse(false)) {
```
Comment (aloubyansky) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:270]: Could you please change the second part to `project.getParentFile() != null && project.getParentFile().exists()`? It's an lot more simpler and efficient expression.

**Review by aloubyansky — CHANGES_REQUESTED** [review]
Thanks a lot @famod. Just a minor change request.

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java (hunk 1 above)
Comment (famod) [reply devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:270]: I prefer the more DRY `Optional` approach but sure, I can (and will) change it.

Comment (famod) [reply devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: I went with a slightly different approach with just one negation instead of two.

**Review by aloubyansky — APPROVED** [review]
Perfect, thanks!

**Pull request merged**